src/solver/conv_ocl_dir2Dfwd.cpp

```diff
@@ -484,12 +489,191 @@ ConvSolution ConvOclDirectFwd::GetSolution(const ConvolutionContext& ctx,
-                                                const LegacyPerformanceConfig& config) const
+namespace fusion {
+
+PerformanceConfigConvOclDirectFwdFused
+ConvOclDirectFwdFused::Search(const FusionContext& problem,
+                              const AnyInvokeParams& invoke_params) const
 {
-    ConvSolution result = BaseGetSolution(ctx, problem, config);
+    const auto& conv_ctx = problem.GetConvContext(0, conv::Direction::Forward);
+    const auto legacy    = ConvOclDirectFwd{};
+    return legacy.Search(conv_ctx, invoke_params);
+}
+
+bool ConvOclDirectFwdFused::IsApplicable(const FusionContext& context) const
+{
+    const auto& desc = *context.problem.fusion_plan_desc;
+    if(desc.op_map.empty())
+    {
+        MIOPEN_THROW("No operators added to fusion plan");
+    }
+    // check the sequence of prims
+    if(desc.op_map.size() > 4)
+        return false;
+    if(desc.op_map[0]->kind() != miopenFusionOpConvForward)
+        return false;
+    if(desc.op_map.size() >= 2)
+    {
+        const auto prim = desc.op_map[1]->kind();
+        if(!(prim == miopenFusionOpBatchNormInference || prim == miopenFusionOpBiasForward ||
+             prim == miopenFusionOpActivForward))
+            return false;
+    }
+    if(desc.op_map.size() >= 3)
+    {
+        const auto prim = desc.op_map[2]->kind();
+        if(!(prim == miopenFusionOpActivForward || prim == miopenFusionOpBatchNormInference))
+            return false;
+    }
+    if(desc.op_map.size() == 4)
+    {
+        const auto prim = desc.op_map[3]->kind();
+        if(!(prim == miopenFusionOpActivForward))
+            return false;
+    }
+    const auto conv_prob = context.problem.GetConvProblem(0, conv::Direction::Forward);
+    if(!conv_prob.IsFp32())
+        return false;
+    const auto base = ConvOclDirectFwd{};
+    return base.IsApplicable(context.GetConvContext(0, conv::Direction::Forward));
+}
+
+ConvSolution
+ConvOclDirectFwdFused::GetSolution(const FusionContext& context,
+                                   const PerformanceConfigConvOclDirectFwdFused& config) const
+{
+    const auto conv_ctx = context.GetConvContext(0, conv::Direction::Forward);
+    ConvSolution result = BaseGetSolution(conv_ctx, conv_ctx.problem, config);
+
+    if(result.construction_params.size() != 1)
+        MIOPEN_THROW("ConvOclDirectFwdFused expects only one kernel");
+
+    auto& kernel_info       = result.construction_params[0];
+    kernel_info.kernel_file = "MIOpenConvDirBatchNormActiv.cl";
+    kernel_info.kernel_name = "MIOpenConvUniBatchNormActiv";
+    const auto& desc        = *context.problem.fusion_plan_desc;
+
+    const int bias_idx  = GetOpIdx(desc.op_map, miopenFusionOpBiasForward);
+    const int activ_idx = GetOpIdx(desc.op_map, miopenFusionOpActivForward);
+    const int bn_idx    = GetOpIdx(desc.op_map, miopenFusionOpBatchNormInference);
+
+    if(bias_idx != -1)
+        kernel_info.comp_options += " -DMLO_CONV_BIAS=" + std::to_string(1);
```

Comment:
Please use `KernelBuildParams` in all new code.

Reply:
Done